src/GraphQL.Conventions/Adapters/Engine/GraphQLExecutor.cs

```diff
@@ -93,9 +95,28 @@ public IGraphQLExecutor<ExecutionResult> WithCancellationToken(CancellationToken
             return this;
         }
 
+        [Obsolete("Please use the WithDependencyInjector overload that accepts an IServiceProvider instance.")]
         public IGraphQLExecutor<ExecutionResult> WithDependencyInjector(IDependencyInjector injector)
         {
-            _dependencyInjector = injector;
+            _serviceProvider = new DependencyInjectorWrapper(injector);
+            return this;
+        }
+
+        private class DependencyInjectorWrapper : IServiceProvider
+        {
+            private readonly IDependencyInjector _injector;
+
+            public DependencyInjectorWrapper(IDependencyInjector injector)
+            {
+                _injector = injector;
+            }
+
+            public object GetService(Type serviceType) => _injector.Resolve(serviceType.GetTypeInfo());
+        }
+
+        public IGraphQLExecutor<ExecutionResult> WithDependencyInjector(IServiceProvider serviceProvider)
```

Comment:
What do you think about having an alias named `WithServiceProvider()` with an `Obsolete` flag steering the user away from using the `WithDependencyInjector()` methods altogether?

Reply:
Sounds good